packages/apify/package.json

```diff
@@ -83,5 +83,9 @@
         "puppeteer": {
             "optional": true
         }
+    },
+    "devDependencies": {
+        "@types/mime-types": "^2.1.1",
+        "@types/stream-json": "^1.7.1"
```

Comment:
dev deps should go to the root package.json

Reply:
Thanks for reminding me to ask you about this! technically, @types dependencies should be included as full dependencies, not devDeps (so users of the library won't get TS compile errors when using our library, nor will they have to enable skipLibCheck). Should we do that? Or keep them as devDeps?

Reply:
> technically, @types dependencies should be included as full dependencies

Not really, only if they are exposed to the public API - so answering with a question:

Are those types exposed to public API? :]

Reply:
_Yes_, but not directly. It's still types that tsc will verify when you build your own project, and if the types are missing, and you don't have skipLibCheck on, your project won't compile. 😢 
I've had this issue in a separate project, where the missing `@types/ws` would cause constant compile errors for the users because our types used types from `@types/ws`.

Reply:
best thing you can do is to try, generally it should not be needed

if the build would fail, we want them there. ideally this should be caught by the ts-check workflow, as that is its sole purpose.

Reply:
Well the ts-check workflow won't catch this right now due to the many build errors + it installing all dependencies, not just production ones. Do we want to change the second one to install just production types?

Reply:
>  it installing all dependencies, not just production ones

It installs just the SDK in a separate project, the previous step that installs SDK deps should not be relevant. If it is like that, the whole workflow does not make sense, but I kinda doubt it - we install SDK deps so we can yarn link it, but the check itself happens in different project. I'd guess yarn linking the SDK should not have any effect on the dependencies.

Reply:
but maybe it does work it the way you describe it - please test it and ideally make it work only with direct deps if possible. or we can remove that workflow, as i dont see any value if it wont change those issues

Reply:
It's not a different project sadly, it's all happening in the same base folder... We should probably have a separate repository for the testing of sdk, or similar? What are your thoughts on it?

Reply:
its a different folder with its own package.json so it should ideally have its own node_modules - that was the idea behind it. i would rather not use separate repo to test it - it has to happen before publishing things, so there would still have to be some kind of yarn link magic (or patch-package).

Reply:
maybe it would be enough to reinstall the SDK deps - we need dev deps so we can build the TS code, so we can then yarn link it, but after that we could drop the SDK node modules folder before we yarn link it to the ts-check project